server/src/test/java/org/elasticsearch/repositories/InvalidRepositoryTests.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.elasticsearch.cluster.metadata.RepositoryMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.snapshots.SnapshotId;
+import org.elasticsearch.test.ESTestCase;
+
+public class InvalidRepositoryTests extends ESTestCase {
+
+    private InvalidRepository repository = new InvalidRepository(
+        new RepositoryMetadata("name", "type", Settings.EMPTY),
+        new RepositoryException("name", "failed to create repository")
+    );
+
+    public void testShouldThrowWhenGettingMetadata() {
+        expectThrows(RepositoryException.class, () -> repository.getSnapshotGlobalMetadata(new SnapshotId("name", "uuid")));
+    }
+
+    public void testShouldNotThrowWhenApplyingLifecycleChanges() {
+        repository.start();
+        repository.stop();
+    }
+
+    public void testShouldNotThrowWhenClosingToAllowRemovingRepo() {
+        repository.close();
+    }
```

Comment:
I think the lifecycle things are covered adequately by the tests that use this repo so we can drop these.